api/app/tests/apps/api/test_api_v1_ml_predict_endpoint.py

```diff
@@ -0,0 +1,138 @@
+            text='For testing purposes we translate this category',
+            created_by='someone@example.com',
+        )
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict')
+    def test_predict(self, patched):
+        patched.return_value = self.link_test_subcategory
+
+        data = {'text': 'Give me the subcategory'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+
+        self.assertEqual(response_json['_links']['self']['href'], self.link_test_subcategory)
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict')
+    def test_predict_translated(self, patched):
+        patched.return_value = self.link_test_subcategory_translated
+
+        data = {'text': 'Give me the subcategory, because of translations'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+
+        # This should be the translated category URL
+        self.assertEqual(response_json['_links']['self']['href'], self.link_test_subcategory)
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict')
+    def test_predict_overig(self, patched):
+        patched.return_value = self.link_test_subcategory_overig
+
+        data = {'text': 'deze test resulteert in de overige categorie'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+
+        self.assertEqual(response_json['_links']['self']['href'], self.link_test_subcategory_overig)
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict')
+    def test_predict_overig_none(self, patched):
+        patched.return_value = None
+
+        data = {'text': 'deze test resulteert in de overige categorie'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 200)
+
+        response_json = response.json()
+
+        self.assertEqual(response_json['_links']['self']['href'], self.link_test_subcategory_overig)
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict',
+           side_effect=Http404)
+    def test_predict_404(self, patched):
+        data = {'text': '404'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 404)
+
+    @patch('signals.apps.api.v1.public.views.MLPredictCategoryView._ml_predict',
+           side_effect=APIException)
+    def test_predict_httpresponseservererror(self, patched):
+        data = {'text': '500'}
+        response = self.client.get(self.endpoint, data=data, format='json')
+
+        self.assertEqual(response.status_code, 500)
```

Comment:
Should we be producing 500s?

Reply:
Nope will only happen if the ML tool produces an error in the 5XX range